src/shared/bus-polkit.c

```diff
@@ -176,6 +178,8 @@ typedef struct AsyncPolkitQuery {
 
         Hashmap *registry;
         sd_event_source *defer_event_source;
+
+        char **authorized_actions;
```

Comment:
hmm, so the details need to be tracked per action.

Hence I guess we need a new structure AsyncPolkitQueryAction which consist of action + details fields, and which we can maintain a linked list of. And each time bus_verify_polkit_async() is invoked we go through the list in full, and compare both the action *and* the details for each, and if we do not find an authorized action for the combination of both we'll ask polkit.

Reply:
Right, that's what I've already done in v3. In addition, I've disposed of the delayed parsing of the polkit reply; it made sense in the original code, but IMHO it only makes the code awkward now. So the polkit response is parsed when in arrives and `AsyncPolkitQuery` contains just the results: a list of authorized actions, a possible denied action, and a possible action that caused an error reply.